src/py/mat3ra/made/tools/convert/__init__.py

```diff
@@ -115,6 +117,10 @@ def from_pymatgen(structure: Union[Structure, Interface]) -> Dict[str, Any]:
                 interface_props[key] = str(value)
         metadata["interface_properties"] = json.loads(json.dumps(interface_props, cls=NumpyNDArrayRoundEncoder))
 
+        interface_labels = list(map(lambda s: INTERFACE_LABELS_MAP[s.properties["interface_label"]], structure.sites))
+
+    basis["labels"] = map_array_to_array_with_id_value(interface_labels, remove_none=True) if interface_labels else []
```

Comment:
Isolate to `extract_labels_from_pymatgen_structure` with something like

```py
def extract_labels_from_pymatgen_structure(structure)
   if structure is interface:
        ...get interface labels here
   otherwise, just get other labels
```